network/src/protocols/disconnect_message.rs

```diff
@@ -27,6 +28,36 @@ impl ServiceProtocol for DisconnectMessageProtocol {
                 "Received disconnect message from peer={}: {}",
                 session_id, message
             );
+
+            // check if baned message and metric banned info
+            if message.starts_with("Ban for") && message.ends_with("seconds") {
```

Comment:
Refer to the `NetworkState::ban_session` here to help code readers double check the message.

Reply:
We can add ban reason in the message in `NetworkState::ban_session`. Here we should handle backward compatibility. It's OK when the ban message has no reason, but if it does, we can record it.